sim/battle.ts

```diff
@@ -2316,6 +2316,14 @@ export class Battle {
 				pokemon.isActive = false;
 				pokemon.isStarted = false;
 				delete pokemon.terastallized;
+				if (pokemon.preFormeChangeInfo) {
+					pokemon.setSpecies(pokemon.preFormeChangeInfo.species);
+					pokemon.baseSpecies = pokemon.preFormeChangeInfo.species;
```

Comment:
I don't think we can do this for every species. For example, Palafin doesn't go back to Zero forme when it faints.

Reply:
I was trying to avoid adding a property in pokedex or something, but it seems like the solution is to either do that or hardcode a list here similar to the disguise/iceface hardcode